File: polymorphism.py
class Car:
    def __init__(self, brand, model, year):
        self.brand = brand
        self.model = model
        self.year = year

    def show_details(self):
        print(f'Brand: {self.brand}\nModel: {self.model}\nYear: {self.year}')

    def fun(self):
       print("Car can run")



class Toyota(Car):
    def __init__(self, model, year, price):
        self.price = price
        super().__init__('Toyota', model, year)

    def fun(self):
        super().fun()
        print("toyta can run")




car1 = Toyota('Corola', 2010, 25000)
car1.show_details()
car1.fun()

File: polymorphism.cpp
#include <iostream>
#include <string>
using namespace std;

class Car {
public:
    string brand;
    string model;
    int year;
    
    Car(string brand, string model, int year) {
        this->brand = brand;
        this->model = model;
        this->year = year;
    }

    void show_details() const {
        cout << "Brand: " << brand << endl;
        cout << "Model: " << model << endl;
        cout << "Year: " << year << endl;
    }

    void fun(){
        cout << "Car can run" << endl;
    }
};

class Toyota : public Car {
public:
    double price;
    
    Toyota(string model, int year, double price) : Car("Toyota", model, year) {
        this->price = price;
    }

    void fun(){
        cout << "Toyota can run" << endl;
    }
};

int main() {
    Toyota car1("Corolla", 2010, 25000);
    car1.show_details();
    car1.fun();

    return 0;
}
